### PR TITLE
QT: Implement themed icons

### PR DESCRIPTION
**Pull request by BreadFish64**
Uses the implementation described here:
http://mithatkonar.com/wiki/doku.php/qt/icons#custom_theme_using_qiconfromtheme

Adds a simple way to use different icons for different themes. This PR does not add any such icons, but it will be useful for a later PR I have planned and @jroweboy has also expressed interest in using them.
New icons can be added in the icons folder and qrc file under their respective themes (default for default icons). Icons are accessed using QIcon::fromTheme().

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/3554)

### CODE REVIEW

File: dist/qt_themes/qdarkstyle/style.qrc [hunk 1]
```diff
@@ -1,4 +1,7 @@
 <RCC>
+  <qresource prefix="icons">
+    <file alias="index.theme">icons/index.theme</file>
```
Comment (jroweboy) [inline dist/qt_themes/qdarkstyle/style.qrc]: What happens when dark theme is missing an icon? Does it just fall back to default theme?

File: dist/qt_themes/default/icons/index.theme [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Icon Theme]
+Name=default
```
Comment (jroweboy) [inline dist/qt_themes/default/icons/index.theme]: doesn't qt use "default" as the default theme name? Is it fine to add our custom theme with the same name?

(i'm scared it'll break linux as the themes are used on there, but i don't think *citra* uses any themed icons anyway)

File: dist/qt_themes/qdarkstyle/style.qrc (hunk 1 above)
Comment (BreadFish64) [reply dist/qt_themes/qdarkstyle/style.qrc]: yup, themes should just inherit icons from default if unavailable: https://github.com/citra-emu/citra/pull/3554/files#diff-1f2995380d219c0a4487fd67307998d9R4

File: dist/qt_themes/default/icons/index.theme (hunk 2 above)
Comment (BreadFish64) [reply dist/qt_themes/default/icons/index.theme]: tested by @daniellimws , doesn't seem to be an issue

**Comment by jroweboy** [comment]
Did you test that it will actually load custom icons from dark theme for when we actually use it? (I know that the fallback works)

**Comment by BreadFish64** [comment]
It seems there was an issue with not setting the theme search path, that should be fixed now:
image
image

**Pull request merged**